sycl/include/sycl/ext/oneapi/matrix/matrix-hip.hpp

```diff
@@ -0,0 +1,389 @@
+          res.data[i] = src[s_idx];
+        }
+      } else {
+        for (int i = 0; i < 4; ++i) {
+          const int s_idx = i + thread_x * stride + thread_y * 4;
+          res.data[i] = src[s_idx];
+        }
+      }
+    } else if constexpr (M == 32 && N == 32) {
+      const auto thread_x = idx % N;
+      const auto thread_y = idx / N;
+
+      if constexpr (Layout == matrix_layout::row_major) {
+        for (int j = 0; j < 4; ++j) {
+          for (int i = 0; i < 4; ++i) {
+            const int s_idx =
+                thread_x + i * stride + thread_y * 4 * stride + j * 8 * N;
+            res.data[i + 4 * j] = src[s_idx];
+          }
+        }
+      } else {
+        for (int j = 0; j < 4; ++j) {
+          for (int i = 0; i < 4; ++i) {
+            const int s_idx = i + thread_x * stride + thread_y * 4 + j * 8;
+            res.data[i + 4 * j] = src[s_idx];
+          }
+        }
+      }
+    }
+  }
+}
+
+template <
+    typename Group, typename S, typename T, size_t M, size_t N,
+    access::address_space Space, access::decorated IsDecorated,
+    typename = std::enable_if_t<std::is_same_v<S, std::remove_const_t<T>>>>
+void load_accumulator_hip(joint_matrix_hip<S, matrix_use::accumulator, M, N,
+                                           matrix_layout::dynamic> &res,
+                          multi_ptr<T, Space, IsDecorated> src, size_t stride,
+                          matrix_layout layout, Group &sg) {
+  static_assert(std::is_same_v<S, int32_t> || std::is_same_v<S, float> ||
+                    std::is_same_v<S, double>,
+                "Unsupported matrix type!");
+
+  if (layout == matrix_layout::row_major)
+    load_accumulator_layoutT<matrix_layout::row_major>(res, src, stride, sg);
+  else
+    load_accumulator_layoutT<matrix_layout::col_major>(res, src, stride, sg);
+}
+
+template <typename Group, typename S, typename T, size_t M, size_t N,
+          matrix_use Use, matrix_layout Layout, access::address_space Space,
+          access::decorated IsDecorated,
+          typename = typename std::enable_if_t<
+              (Layout == matrix_layout::row_major ||
+               Layout == matrix_layout::col_major) &&
+              std::is_same_v<S, std::remove_const_t<T>>>>
+void load_multiplicand_hip(joint_matrix_hip<S, Use, M, N, Layout> &res,
+                           multi_ptr<T, Space, IsDecorated> src, size_t stride,
+                           Group &sg) {
+  static_assert(std::is_same_v<S, half> || std::is_same_v<S, bfloat16> ||
+                    std::is_same_v<S, int8_t> || std::is_same_v<S, double>,
+                "Unsupported matrix type!");
+
+  const auto idx = sg.get_group_linear_id() * sg.get_local_range()[0] +
+                   sg.get_local_linear_id();
+
+  if constexpr (std::is_same_v<S, double>) {
+    if constexpr (Layout == matrix_layout::row_major) {
+      res.data[0] = src[idx];
+    } else if constexpr (Layout == matrix_layout::col_major) {
```

Comment:
nit: this `else if` is redundant since you have the `std::enable_if_t<
              (Layout == matrix_layout::row_major ||
               Layout == matrix_layout::col_major)` above I think.

Reply:
Makes sense.